// Copyright 2022 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

syntax = "proto2";

option optimize_for = LITE_RUNTIME;

package apps;

// Describes the type of app being observed. Should be kept in sync with the one
// in components/services/app_service/public/cpp/app_types.h.
enum ApplicationType {
  APPLICATION_TYPE_UNKNOWN = 0;
  APPLICATION_TYPE_ARC = 1;
  APPLICATION_TYPE_BUILT_IN = 2;
  APPLICATION_TYPE_CROSTINI = 3;
  APPLICATION_TYPE_CHROME_APP = 4;
  APPLICATION_TYPE_WEB = 5;
  APPLICATION_TYPE_MAC_OS = 6;
  APPLICATION_TYPE_PLUGIN_VM = 7;
  APPLICATION_TYPE_STANDALONE_BROWSER = 8;
  APPLICATION_TYPE_REMOTE = 9;
  APPLICATION_TYPE_BOREALIS = 10;
  APPLICATION_TYPE_SYSTEM_WEB = 11;
  APPLICATION_TYPE_STANDALONE_BROWSER_CHROME_APP = 12;
  APPLICATION_TYPE_EXTENSION = 13;
  APPLICATION_TYPE_STANDALONE_BROWSER_EXTENSION = 14;
  APPLICATION_TYPE_BRUSCHETTA = 15;
}

// Describes the app installation source. Should be kept in sync with the one
// in components/services/app_service/public/cpp/app_types.h.
enum ApplicationInstallSource {
  APPLICATION_INSTALL_SOURCE_UNKNOWN = 0;
  APPLICATION_INSTALL_SOURCE_SYSTEM = 1;
  APPLICATION_INSTALL_SOURCE_SYNC = 2;
  APPLICATION_INSTALL_SOURCE_PLAY_STORE = 3;
  APPLICATION_INSTALL_SOURCE_CHROME_WEB_STORE = 4;
  APPLICATION_INSTALL_SOURCE_BROWSER = 5;
}

// Describes the app installation reason. Should be kept in sync with the one
// in components/services/app_service/public/cpp/app_types.h.
enum ApplicationInstallReason {
  APPLICATION_INSTALL_REASON_UNKNOWN = 0;
  APPLICATION_INSTALL_REASON_SYSTEM = 1;
  APPLICATION_INSTALL_REASON_POLICY = 2;
  APPLICATION_INSTALL_REASON_OEM = 3;
  APPLICATION_INSTALL_REASON_DEFAULT = 4;
  APPLICATION_INSTALL_REASON_SYNC = 5;
  APPLICATION_INSTALL_REASON_USER = 6;
  APPLICATION_INSTALL_REASON_SUB_APP = 7;
  APPLICATION_INSTALL_REASON_KIOSK = 8;
  APPLICATION_INSTALL_REASON_COMMAND_LINE = 9;
}

// Describes the app installation time depending on the app lifecycle. Should
// be kept in sync with the one in
// chrome/browser/apps/app_service/metrics/app_platform_metrics.h.
enum ApplicationInstallTime {
  APPLICATION_INSTALL_TIME_UNKNOWN = 0;
  APPLICATION_INSTALL_TIME_INIT = 1;
  APPLICATION_INSTALL_TIME_RUNNING = 2;
}

// Describes the app launch source. Should be kept in sync with the ones in
// components/services/app_service/public/cpp/app_launch_util.h
enum ApplicationLaunchSource {
  APPLICATION_LAUNCH_SOURCE_UNKNOWN = 0;
  APPLICATION_LAUNCH_SOURCE_APP_LIST_GRID = 1;
  APPLICATION_LAUNCH_SOURCE_APP_LIST_GRID_CONTEXT_MENU = 2;
  APPLICATION_LAUNCH_SOURCE_APP_LIST_QUERY = 3;
  APPLICATION_LAUNCH_SOURCE_APP_LIST_QUERY_CONTEXT_MENU = 4;
  APPLICATION_LAUNCH_SOURCE_APP_LIST_RECOMMENDATION = 5;

  APPLICATION_LAUNCH_SOURCE_PARENTAL_CONTROLS = 6;
  APPLICATION_LAUNCH_SOURCE_SHELF = 7;
  APPLICATION_LAUNCH_SOURCE_FILE_MANAGER = 8;
  APPLICATION_LAUNCH_SOURCE_LINK = 9;
  APPLICATION_LAUNCH_SOURCE_OMNIBOX = 10;

  APPLICATION_LAUNCH_SOURCE_CHROME_INTERNAL = 11;
  APPLICATION_LAUNCH_SOURCE_KEYBOARD = 12;
  APPLICATION_LAUNCH_SOURCE_OTHER_APP = 13;
  APPLICATION_LAUNCH_SOURCE_MENU = 14;
  APPLICATION_LAUNCH_SOURCE_INSTALLED_NOTIFICATION = 15;
  APPLICATION_LAUNCH_SOURCE_TEST = 16;
  APPLICATION_LAUNCH_SOURCE_ARC = 17;
  APPLICATION_LAUNCH_SOURCE_SHARESHEET = 18;
  APPLICATION_LAUNCH_SOURCE_RELEASE_NOTES_NOTIFICATION = 19;
  APPLICATION_LAUNCH_SOURCE_FULL_RESTORE = 20;
  APPLICATION_LAUNCH_SOURCE_SMART_TEXT_CONTEXT_MENU = 21;
  APPLICATION_LAUNCH_SOURCE_DISCOVER_TAB_NOTIFICATION = 22;
  APPLICATION_LAUNCH_SOURCE_MANAGEMENT_API = 23;
  APPLICATION_LAUNCH_SOURCE_KIOSK = 24;
  APPLICATION_LAUNCH_SOURCE_COMMAND_LINE = 25;
  APPLICATION_LAUNCH_SOURCE_BACKGROUND_MODE = 26;
  APPLICATION_LAUNCH_SOURCE_NEW_TAB_PAGE = 27;
  APPLICATION_LAUNCH_SOURCE_INTENT_URL = 28;
  APPLICATION_LAUNCH_SOURCE_OS_LOGIN = 29;
  APPLICATION_LAUNCH_SOURCE_PROTOCOL_HANDLER = 30;
  APPLICATION_LAUNCH_SOURCE_URL_HANDLER = 31;
  APPLICATION_LAUNCH_SOURCE_LOCK_SCREEN = 32;
  APPLICATION_LAUNCH_SOURCE_APP_HOME_PAGE = 33;
  APPLICATION_LAUNCH_SOURCE_REPARENTING = 34;
  APPLICATION_LAUNCH_SOURCE_PROFILE_MENU = 35;
  APPLICATION_LAUNCH_SOURCE_SYSTEM_TRAY_CALENDAR = 36;
}

// Describes the app uninstall source. Should be kept in sync with
// components/services/app_service/public/cpp/app_types.h.
enum ApplicationUninstallSource {
  APPLICATION_UNINSTALL_SOURCE_UNKNOWN = 0;
  APPLICATION_UNINSTALL_SOURCE_APP_LIST = 1;
  APPLICATION_UNINSTALL_SOURCE_APP_MANAGEMENT = 2;
  APPLICATION_UNINSTALL_SOURCE_SHELF = 3;
  APPLICATION_UNINSTALL_SOURCE_MIGRATION = 4;
}
